pages/blog/data-cleaning-null-aggregate-values.mdx

```diff
@@ -0,0 +1,354 @@
+
+We’re going to walk through this LTV example from beginning to end looking at different options -- some that work well, some that work poorly, and some that won’t work at all.
+
+To start, we have a `users` table and a `purchases` table.
+
+users:
+
+<table style={{ textAlign: "center" }}>
+  <thead>
+    <tr>
+      <th style={{ border: "1px solid black", padding: "4px" }}> user_id </th>
+    </tr>
+  </thead>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>1</td>
+  </tr>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>2</td>
+  </tr>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>3</td>
+  </tr>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>4</td>
+  </tr>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>5</td>
+  </tr>
+</table>
+<br />
+purchases:
+
+<table style={{ textAlign: "center" }}>
+  <thead>
+    <tr>
+      <th style={{ border: "1px solid black", padding: "4px" }}> user_id </th>
+      <th style={{ border: "1px solid black", padding: "4px" }}> price </th>
+    </tr>
+  </thead>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>1</td>
+    <td style={{ border: "1px solid black", padding: "4px" }}>79.36</td>
+  </tr>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>3</td>
+    <td style={{ border: "1px solid black", padding: "4px" }}>42.97</td>
+  </tr>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>1</td>
+    <td style={{ border: "1px solid black", padding: "4px" }}>33.14</td>
+  </tr>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>2</td>
+    <td style={{ border: "1px solid black", padding: "4px" }}>16.48</td>
+  </tr>
+  <tr>
+    <td style={{ border: "1px solid black", padding: "4px" }}>4</td>
+    <td style={{ border: "1px solid black", padding: "4px" }}>96.41</td>
+  </tr>
+</table>
+<br />
+
+As you can see, the users table has 5 unique users and the purchases table has 5 purchases from users 1, 2, 3, and 4. User 5 is not present in the purchases table. Now that we have our data, it’s time to create our calculated column. Our query is being called by a function that passes it a list of user_ids that can be used in the query.
+
+### Option 1: Using SQL
+
+#### Using `COALESCE`
+
+`COALESCE` is a keyword used by Postgres and other SQL dialects to return the first non-null value in a set. Because of this, it’s a common keyword for data cleaning. Let’s look closer at the following query.
+
+```sql
+SELECT
+	COALESCE(SUM(purchases.price), 0) as __result,
+	user_id as __pk
+FROM purchases
+WHERE purchases.user_id
+	IN (1, 2, 3, 4, 5)
+GROUP BY __pk
+ORDER BY __pk;
```

Comment:
Using `__pk` is something we do at Grouparoo, but doesn't seem necessary to post's content.  Maybe just stick with `user_id` and don't rename the column?  The same goes for `__result` - perhaps it would be clearer to call that `ltv`